### ISSUE TITLE
Requesting Seurat v4.9.9.9060

### ISSUE THREAD

**Issue by Victor21v** [satijalab/seurat#9499]
I would like to inquire whether anyone can provide me with the beta version of Seurat 4.9.9.9060. I am currently attempting to run the SketchData() function on my dataset of 1.3 million cells, but the process is taking a considerable amount of time to complete.

As mentioned in issue #8127, this issue appears to have been introduced with Seurat v5, and it seems that the problem could be resolved by using a previous version. However, I have been unable to locate this version for Linux systems.

Any help or guidance would be greatly appreciated. Thank you in advance!

**Comment by inofechm** [comment]
[seurat-3f907ea681893187bd623d704e6ceaf4870a891b.zip](https://github.com/user-attachments/files/17924213/seurat-3f907ea681893187bd623d704e6ceaf4870a891b.zip) I was able track down the commit for this version if you want to install directly from here.

**Comment by longmanz** [comment]
Thank you @inofechm for sharing the older version. There is a bug with the sketchData and we are working on it. Sorry for the inconvenience.

**Comment by Victor21v** [comment]
> Thank you [@inofechm](https://github.com/inofechm) for sharing the older version. There is a bug with the sketchData and we are working on it. Sorry for the inconvenience.

Thanks for the response, was it fixed with the last seurat update (Seurat 5.2.1)?

**Comment by sentisci** [comment]
> > Thank you [@inofechm](https://github.com/inofechm) for sharing the older version. There is a bug with the sketchData and we are working on it. Sorry for the inconvenience.
> 
> Thanks for the response, was it fixed with the last seurat update (Seurat 5.2.1)?

I dont think it is fixed in Seurat 5.2.1 I am still facing the same issue. It works fine when the data comprises of millions of cells from only few libraries only but chokes when there are large number of libraries to begin with.

**Comment by longmanz** [comment]
Hi, 
Sorry for the late response. This is still a bug (we have confirmed and fixed it) and will be updated in the next release. Will keep you updated

**Comment by YoukaiFromAccounting** [comment]
Hello @Victor21v @sentisci , thank you for bringing this to our attention, and apologies for the delay. Could you try installing the latest version of Seurat (updated this past week), which should include an adjustment in the LeverageScore function logic within SketchData? Additionally, could you try not specifying VariableFeatures in your SketchData call (the function should automatically retrieve the features from the previous FindVariableFeatures call)

You can install the latest version of Seurat in RStudio via:
```{R}
devtools::install_github("satijalab/seurat", ref = "main")
```

Alternatively, you can clone the repository in Command Line and install it from a local directory:
```{R}
git clone https://github.com/satijalab/seurat.git
cd seurat
git checkout main
```

```{R}
devtools::install_local("path/to/seurat")
```

Please let us know if the error persists.

**Comment by sentisci** [comment]
I will try and let you know

**Comment by YoukaiFromAccounting** [comment]
Closing this for now as there hasn't been any recent activity. If this is still causing trouble or you'd like to revisit it, feel free to reopen.
